### PR TITLE
BUG: Fix metadata propagation in df.corr and df.cov, GH28283

### PR DESCRIPTION
**Pull request by yuanx749**
- [x] xref #28283 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v1.6.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/generic/test_finalize.py [hunk 1]
```diff
@@ -197,12 +197,10 @@
     ),
     pytest.param(
         (pd.DataFrame, frame_data, operator.methodcaller("corr")),
-        marks=not_implemented_mark,
```
Comment (mroeschke) [inline pandas/tests/generic/test_finalize.py:200]: Could you remove the outer `pytest.param` as well? (Same as below)

Comment (yuanx749) [reply pandas/tests/generic/test_finalize.py:200]: Sure. The one below has another mark, which requires `pytest.param`.

I saw there are some `pytest.param` leftovers in the file. Should I also remove them, or just leave it for future maintenance?

Comment (mroeschke) [reply pandas/tests/generic/test_finalize.py:197]: Got it. We can leave the one below then.

You can leave it for now and tackle it in a follow up PR if you'd like.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @yuanx749

**Pull request merged**